src/kirby/components/modal/services/modal.controller.ts

```diff
@@ -36,19 +40,34 @@ export class ModalController implements IModalController {
     });
   }
 
-  public registerModal(modal: { close: (data?: any) => {} }): void {
+  public openActionSheet(
+    config: ActionSheetConfig,
+    vcRef: ViewContainerRef,
+    onCloseModal?: (data?: any) => void
+  ): void {
+    this.actionSheetHelper
+      .showActionSheet(config, vcRef, this.registerModalWrapper)
```

Comment:
Just bind the method to `this`:
```suggestion
      .showActionSheet(config, vcRef, this.register.bind(this))
)
```

Reply:
this is cool ! fixed